packages/core/src/compiler.ts

```diff
@@ -171,7 +171,12 @@ export default class Compiler {
         this.setRuntimeFn('plural');
         break;
 
-      case 'function':
+      case 'function': {
+        const formatter = this.options.customFormatters?.[token.key];
+        const formattingModuleRequest =
+          formatter && 'module' in formatter ? formatter.module : null;
+        const isLocaleSpecificFormattingModule =
+          typeof formattingModuleRequest === 'function';
```

Comment:
The temporary variable is more confusing than not.
```suggestion
        const isLocaleSpecificFormattingModule = typeof formatter?.module === 'function';
```

Also, "isLocaleSpecificFormattingModule" is an excessively long name for a local variable. With changes below, you can probably drop it rather than renaming it.

Reply:
5bf4e84